src/network/networkd-dhcp4.c

```diff
@@ -147,9 +149,17 @@ static int link_set_dhcp_routes(Link *link) {
 
         n = sd_dhcp_lease_get_routes(link->dhcp_lease, &static_routes);
         if (n == -ENODATA)
-                return 0;
-        if (n < 0)
-                return log_link_warning_errno(link, n, "DHCP error: could not get routes: %m");
+                log_link_info(link, "DHCP: No routes received from DHCP server: %m");
+        else if (n < 0)
+                log_link_warning_errno(link, n, "DHCP error: could not get routes: %m");
+
+        if (n < 0 && r < 0) {
```

Comment:
hmm, i really don't like that you check r here, after all it has possibly been updated in the mean time with lots of other values since it was set to -ENODATA because no routes were discovered.

Why not just use the the test "link->dhcp4_messages == 0" here instead? that's a much more correct and explicit test under the same conditions, or am I wrong?

Reply:
yes that is much cleaner. thanks !